### PR TITLE
PLAN-80 Username/password login

### PR DESCRIPTION
**Pull request by Gailbear**
Login:
- Can log in with a username and password
- Can ask for a password reset email
- Can click the link in the password reset email and reset your password
- Error messaging for when those things don't work
- Terms of Service link
- Privacy Policy link

Development environment improvements:
- livereload
- mailcatcher
- storybook

### CODE REVIEW

File: app/javascript/login/forgot_password.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                this.error.text = LOGIN_INVALID_FIELDS;
+                this.error.visible = true;
+            } else {
+                http.post('/people/password.json', { person: this.person }).then(data => {
+                    this.successfullySent = data.status === 201
+                    if (this.successfullySent) {
+                        this.$router.push('/?alert=reset_sent')
+                    } else {
+                        console.log(data)
+                        this.alert.text = "TODO WRITE THIS COPY EMAIL DIDN'T SEND"
```
Comment (Gailbear) [inline app/javascript/login/forgot_password.vue:63]: - [ ] fix this to use SOMETHING_WENT_WRONG

File: app/javascript/login/forgot_password.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            if(this.form.email.valid === false) {
+                this.error.text = LOGIN_INVALID_FIELDS;
+                this.error.visible = true;
+            } else {
+                http.post('/people/password.json', { person: this.person }).then(data => {
+                    this.successfullySent = data.status === 201
+                    if (this.successfullySent) {
+                        this.$router.push('/?alert=reset_sent')
+                    } else {
+                        console.log(data)
```
Comment (Gailbear) [inline app/javascript/login/forgot_password.vue:62]: - [ ] remove

File: app/javascript/login/forgot_password.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                        this.alert.text = "TODO WRITE THIS COPY EMAIL DIDN'T SEND"
+                        this.alert.visible = true
+                    }
+                }).catch(error => {
+                    this.alert.text = "TODO WRITE THIS COPY EMAIL DIDN'T SEND"
```
Comment (Gailbear) [inline app/javascript/login/forgot_password.vue:67]: - [ ] fix this to use SOMETHING_WENT_WRONG

File: app/javascript/login/login.stories.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            component.onSubmit = (event) => {
+                event.preventDefault();
+                action('onSubmit')('Submitted!')
+            }
+        },
+        template: '<PlanLogin/>'
+    }
+}
+
+export const Default = Template.bind({});
```
Comment (Gailbear) [inline app/javascript/login/login.stories.js:32]: I think these mostly don't work right now, I'll have to look into fixing them later.

File: app/javascript/login/new_password.vue [hunk 5]
```diff
@@ -0,0 +1,115 @@
+<template>
+  <div class="new-password">
+    <h3>Reset Password</h3>
+    <p>In order to protect your account, make sure your password:</p>
+    <ul>
+      <li>Is 6 or more characters</li>
+      <li>Does not contain the word "password"</li>
+      <li>Does not contain your email address</li>
+      <li>Is not one of your most recent passwords</li>
+      <li>Is not a member of this <a>TODO list of common passwords TODO</a></li>
```
Comment (Gailbear) [inline app/javascript/login/new_password.vue:10]: - [ ] Link to list of common passwords

File: app/javascript/login/new_password.vue [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    </ul>
+    <p>
+      Optionally, we recommend you follow the
+      <a>TODO Correct Horse Battery Staple TODO</a> paradigm.
```
Comment (Gailbear) [inline app/javascript/login/new_password.vue:14]: - [ ] link to explanation

File: app/views/devise/sessions/new.html.erb [hunk 7]
```diff
@@ -1,4 +1,11 @@
+<%= stylesheet_pack_tag 'login_vue' %>
+<%= javascript_pack_tag 'login_vue' %>
 
+<div id="login-app">
+  <login-screen></login-screen>
+</div>
+
+<!--
```
Comment (Gailbear) [inline app/views/devise/sessions/new.html.erb:8]: - [ ] delete old login page comment

**Comment by Gailbear** [comment]
- [ ] format everything

File: app/javascript/constants/strings.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+    // login page
+    LOGIN_401: "You have an incorrect username or password.",
+    LOGIN_MISSING_PASSWORD: "You must enter a password.",
+    LOGIN_INVALID_FIELDS: "You must provide all of your login information.",
+    LOGIN_MISSING_EMAIL: "You must enter an email address.",
+    LOGIN_NOT_AN_EMAIL: "Email address needs to be in the correct format e.g. xyz@test.com",
```
Comment (arbradley) [inline app/javascript/constants/strings.js:7]: For consistency we should make that "Your email address needs " etc

**Review by arbradley — COMMENTED** [review]
strings fine other then see my comment

File: app/javascript/constants/strings.js (hunk 8 above)
Comment (Gailbear) [reply app/javascript/constants/strings.js:7]: fixing

**Pull request merged**